docs/design/classes.md

```diff
@@ -910,6 +915,71 @@ the `me` parameter must be in the same list in square brackets `[`...`]`. The
 `me` parameter may appear in any position in that list, as long as it appears
 after any names needed to describe its type.
 
+#### Extension methods
+
+A function can have a `me` parameter in its implicit parameter list regardless
+of the scope in which it is declared. Any function with a `me` parameter is a
+[method](#methods), but only those functions declared inside a class are member
+functions. Methods that are neither member functions nor
+[`impl` members](generics/details.md#qualified-member-names) nor adapter
+members, and methods whose `me` parameter does not accept an argument of type
+`Self` or, for `addr me`, `Self*`, are called
+[_extension methods_](https://en.wikipedia.org/wiki/Extension_method).
+
+Extension methods are not found by the name lookup used in
+[simple member access](expressions/member_access.md), so compound member access
+notation must be used to call an extension method:
+
+```
+class Rectangle {
+  var width: i32;
+  var height: i32;
+}
+fn Area[me: Rectangle]() -> i32 { return me.width * me.height; }
+
+var r: Rectangle = {.width = 6, .height = 9};
+// ✅ Finds function `Area` defined above.
+var answer: i32 = r.(Area)();
+// ❌ No function `Area` declared in `Rectangle`.
+var bad_answer: i32 = r.Area();
+```
+
+**Note:** It's also possible to alias an extension method into a class and call
+it using simple member access notation.
+
+The type of `me` is not required to be a class type:
```

Comment:
I think `i32` is going to be a class type though....

Maybe "There are no restrictions on the types that can be used with `me` parameters."?